src/transport_ofi.c

```diff
@@ -1562,6 +1562,7 @@ int query_for_fabric(struct fabric_info *info)
                 multirail_fabric_list_tail = cur_fabric;
             }
         }
+        multirail_fabric_list_tail->next = NULL;
```

Comment:
This can generate a null pointer assignment if it is not assigned within the for loop above.

Reply:
Looks like it was the cause of the test failure. Thanks for catching it!